File: app/src/main/java/com/irishinterest/irishinterest/network/NetworkAdapter.java
package com.irishinterest.irishinterest.network;

import android.content.Context;

import com.irishinterest.irishinterest.network.api.irishInterest.IrishInterestAPI;
import com.irishinterest.irishinterest.network.api.irishInterest.observer.GuiObserver;

/**
 * This is the network adapter class. All network requests go through here.
 * At the moment we only have Irish Interest API connection upon initialization.
 */

public class NetworkAdapter {

    private GuiObserver guiObserver;
    private Context context;
    private IrishInterestAPI irishInterestAPI;

    public NetworkAdapter(GuiObserver guiObserver, Context context) {
        this.guiObserver = guiObserver;
        this.context = context;
        initialize();
    }

    private void initialize(){
        this.irishInterestAPI = new IrishInterestAPI(guiObserver, context);
    }
}

File: app/src/main/java/com/irishinterest/irishinterest/network/NetworkAdapter.kt
package com.irishinterest.irishinterest.network

import android.content.Context
import com.irishinterest.irishinterest.network.api.irishInterest.observer.GuiObserver
import com.irishinterest.irishinterest.network.api.irishInterest.IrishInterestAPI

/**
 * This is the network adapter class. All network requests go through here.
 * At the moment we only have Irish Interest API connection upon initialization.
 */
class NetworkAdapter(private val guiObserver: GuiObserver, private val context: Context) {
    private var irishInterestAPI: IrishInterestAPI? = null
    private fun initialize() {
        irishInterestAPI = IrishInterestAPI(guiObserver, context)
    }

    init {
        initialize()
    }
}